lib/ui/painting.dart

```diff
@@ -4252,13 +4252,62 @@ class FragmentShader extends Shader {
   late Float32List _floats;
 
   /// Sets the float uniform at [index] to [value].
+  ///
+  /// All uniforms defined in a fragment shader that are not samplers must be
+  /// set through this method. This includes floats, ints, bools, and vectors.
+  /// The correct index for each uniform is determined by the order of the
+  /// uniforms as defined in the fragment program, ignoring any samplers. For
+  /// data types that are composed of multiple floats such as a vec4, more than
+  /// one call to [setFloat] is required.
+  ///
+  /// For example, given the following uniforms in a fragment program:
+  ///
+  /// ```glsl
+  /// uniform float uScale;
+  /// uniform int uPitch;
```

Comment:
I'm not confident that int (or bool?) uniforms are properly supported yet. We'd at least want `setInt()` and `setBool()` methods, so that folks don't have to set int and bool uniforms using `setFloat()`.

Reply:
They seem to "work" on skia, I'll dig into this a bit more for impeller

Reply:
Yeah this doesn't work when setting fields when we compiler for impellerc. I'll update this to leave them out for now.